superset-frontend/src/SqlLab/components/EditorAutoSync/index.tsx

```diff
@@ -119,6 +126,22 @@ const EditorAutoSync: FC = () => {
     }
   });
 
+  const syncDeletedQueryEditor = useEffectEvent(() => {
```

Comment:
Maybe use `async/await` here? I think it's more readable so feel free to apply the same pattern everywhere.

Reply:
This structure sends delete requests simultaneously, so if implemented with await, additional `Promise.all` handling might be required. Therefore, I intend to keep the current structure.